### PR TITLE
Sort hack/.linted packages

### PR DESCRIPTION
**Pull request by sttts**
Appending to the file by default creates merge conflicts in the submit queue. Sorting will reduce that considerably.

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/31104)

### CODE REVIEW

**Comment by sttts** [comment]
@k8s-bot unit test this issue: #30228

**Comment by dims** [comment]
@sttts : Looks good to me! 👍

**Comment by jfrazelle** [comment]
LGTM

File: hack/verify-golint.sh [hunk 1]
```diff
@@ -82,12 +82,14 @@ if [ ${#linted[@]} -eq 0 ]; then
...
-		echo "The following packages passed golint but are not listed in $linted_file:"
+		echo "Some packages passed golint but are not listed in $linted_file:"
+		echo
 		for p in "${linted[@]}"; do
 			echo "echo $p >> hack/.linted_packages"
 		done
+		echo "sort -o hack/.linted_packages hack/.linted_packages"
 		echo
-		echo 'Please add the following packages to the linted file. You can test via this script and commit the result.'
+		echo 'Please add those packages to the linted file. You can test via this script and commit the result.'
```
Comment (MHBauer) [inline hack/verify-golint.sh]: Say to add them in alphabetical order, or mention that `.linted_packages` is in alphabetical order.

Comment (sttts) [inline hack/verify-golint.sh]: Or even better: check that it is in alphabetical order.

**Comment by sttts** [comment]
@jfrazelle @MHBauer ptal, added a check whether the file is sorted.

**Comment by jfrazelle** [comment]
it seems to be mad in the check :)

**Comment by sttts** [comment]
Self-referential tests ;)

**Comment by jfrazelle** [comment]
this should pass verification though

**Comment by sttts** [comment]
It does locally, interesting.

File: hack/verify-golint.sh [hunk 2]
```diff
@@ -78,16 +94,20 @@ else
...
 	echo 'Success! All packages that should pass lint are listed in the linted file.'
 else
 	{
-		echo "The following packages passed golint but are not listed in $linted_file:"
+		echo "Some packages passed golint but are not listed in hack/.linted_packages."
+		echo "Please add them in alphabetical order:"
+		echo
 		for p in "${linted[@]}"; do
-			echo "echo $p >> hack/.linted_packages"
+			echo "  echo $p >> hack/.linted_packages"
```
Comment (ingvagabund) [inline hack/verify-golint.sh]: The two blanks in front of echo, is it intentional?

Comment (ingvagabund) [inline hack/verify-golint.sh]: I see, it is just a comment, not actual command

**Comment by sttts** [comment]
Found the problem: github merges the PR with master. But on master we have

```
pkg/registry/pod/etcd
pkg/registry/podsecuritypolicy/etcd
```

Which is in the wrong order.

**Comment by sttts** [comment]
Increased priority to get through the queue. Until this is merged, all PRs conflict which change the linted file at the end.

**Comment by ingvagabund** [comment]
re-adding lgtm as there is no change in ``hack/verify-golint.sh`` since the first lgtm. The only thing changed is updated ``hack/.linted_packages`` as a reaction to new PRs that get merged between the first lgtm and now.

**Comment by sttts** [comment]
LANG=C and LANG=de_DE.UTF-8 sort differently :scream:

**Comment by sttts** [comment]
@k8s-bot test this issue: #31199

**Comment by sttts** [comment]
@k8s-bot test this issue: #31262

**Comment by jfrazelle** [comment]
@k8s-bot test this issue: #31262

**Comment by jfrazelle** [comment]
LGTM

File: hack/verify-golint.sh [hunk 3]
```diff
@@ -78,16 +90,20 @@ else
...
 		done
+		echo "  LANG=C sort -o hack/.linted_packages hack/.linted_packages"
 		echo
-		echo 'Please add the following packages to the linted file. You can test via this script and commit the result.'
+		echo 'You can test via this script and commit the result.'
```
Comment (krousey) [inline hack/verify-golint.sh]: Test via this script? Do you mean that they can fix this with the script you echo'd or test that it's fixed with this verify script? Either way, this should change.

Comment (sttts) [inline hack/verify-golint.sh]: Please open a follow-up PR to fix this. I agree that the wording is not really good. It was even worse before this PR:

**Pull request merged**